clarity_ext/domain/analyte.py

```diff
@@ -67,3 +68,12 @@ def updated_rest_resource(self, original_rest_resource, updated_fields):
             _updated_rest_resource.name = self.assigner.register_assign('name', self.name)
         return _updated_rest_resource, self.assigner.consume()
 
+    @property
+    def sample(self):
+        """
+        Returns a single sample for convenience. Throws an error if there isn't exactly one sample.
+
+        NOTE: There can be more than one sample on an Analyte. That's the case with pools.
+        """
+        return utils.single(self.samples)
```

Comment:
The business case being solved: We need to support "pooled" analytes. If you want details, you can check out [this](http://zarlab.cs.ucla.edu/sequencing-with-dna-pools/) article, or ask the lab. Before this commit, this use case hadn't turned up yet, so it supported only one sample.

Rationale behind providing both `samples` and `sample`: A design decision behind the library is that all domain objects should be as easy to use as possible. Where to drag that line can be difficult, but the idea here is that the domain expert (maintaining the extension) can write either `analyte.samples[0]` or `analyte.sample`, depending on what they are expecting. If they do only expect there to be one sample, they don't need to write the guards themselves. The underlying code should take care of throwing an exception if required. (This has the additional benefit that I didn't need to refactor all usage of `sample` already in the codebase when introducing the change.)

It could be argued that the code would allow for there to be no sample at all (returning `None` when zero), but I don't believe that will be needed. It can easily be changed afterwards if required.